src/bootstrap/src/bin/rustc.rs

```diff
@@ -137,6 +137,15 @@ fn main() {
         cmd.args(lint_flags.split_whitespace());
     }
 
+    // Use an untracked env var (untracked by cargo) to pass `-Zon-broken-pipe=kill` to rustc when
+    // this shim is *not* used to build cargo itself, i.e. set `-Zon-broken-pipe=kill` only when
+    // building non-cargo tools.
```

Comment:
We don’t need to mention `cargo` here. Rustc shim doesn’t know anything about `cargo` at this point. It’s more about where we propagate this environment variable.

Reply:
Dropped cargo tracking here and amended the comment to only explain why the flag is conditionally passed.